raiden/raiden_service.py

```diff
@@ -635,6 +651,214 @@ def settle(self, token_address, partner_address):
+                self.raiden.api.open(
+                    self.token_address.encode('hex'),
+                    partner.encode('hex')
+                )
+                self.raiden.api.deposit(
+                    self.token_address.encode('hex'),
+                    partner.encode('hex'),
+                    funding
+                )
+
+    def leave(self, wait_for_settle=True, max_wait=30):
+        """
+        Leave the token network.
+        This implies closing all open channels and optionally wait for settlement.
+        Args:
+            wait_for_settle (bool): should the call only return after setlement
+            max_wait (float): maximum time to wait
+        """
+        with self.lock:
+            open_channels = self.open_channels
+            channel_specs = [(
+                self.token_address.encode('hex'),
+                c.partner_address.encode('hex')) for c in open_channels]
+            pool = Pool(len(channel_specs))
+            pool.imap(lambda c: self.raiden.api.close(*c), channel_specs).join()
+            log.debug("'leave()' closed all channels", num=len(channel_specs))
+
+            if wait_for_settle:
+                timeout = time.time() + max_wait
+                while any(c.state != 'settled' for c in open_channels):
+                    gevent.sleep(1)
+                    if time.time() > timeout:
+                        log.debug(
+                            "timeout while waiting for settlement",
+                            unsettled=len(filter(lambda c: c.state != 'settled', open_channels)),
+                            settled=len(filter(lambda c: c.state == 'settled', open_channels))
+                        )
+                        break
+
+    def _join_channel(self, partner_address, partner_deposit):
+        """Will be called, when we were selected as channel partner by another
+        node. It will fund the channel with up to the partner's deposit, but not
+        more than remaining funds or the initial funding per channel.
+
+        If the connection manager has no funds, this is a noop.
+        """
+        # not initialized
+        if self.funds <= 0:
+            return
+        with self.lock:
+            remaining = self.funds_remaining
+            initial = self.initial_funding_per_partner
+            joining_funds = min(
+                partner_deposit,
+                remaining,
+                initial
+            )
+            if joining_funds <= 0:
+                return
+
+            self.raiden.api.deposit(
+                self.token_address.encode('hex'),
+                partner_address.encode('hex'),
+                joining_funds
+            )
+            log.debug(
+                "joined a channel!",
+                funds=joining_funds,
+                me=pex(self.raiden.address),
+                partner=pex(partner_address)
+            )
+
+    def _retry_connect(self):
```

Comment:
Question: The `_` prefix naming convention for a function signifies a private method? Which means what in our case?

Looking around the PR both `_retry_connect()` and `find_new_partners()` are only called from inside the `ConnectionManager` class.